dbms/src/Storages/Page/V3/tests/gtest_blob_store.cpp

```diff
@@ -1616,4 +1613,143 @@ try
+        ASSERT_TRUE(stat->isReadOnly());
+
+        blob_store.remove({entry_from_write1});
+
+        // new write will create new blob file
+        size_t size_100 = 100;
+        char c_buff[size_100];
+
+        WriteBatch wb;
+        ReadBufferPtr buff = std::make_shared<ReadBufferFromMemory>(const_cast<char *>(c_buff), size_100);
+        wb.putPage(page_id, /* tag */ 0, buff, size_100);
+        auto edit = blob_store.write(wb, nullptr);
+        const auto & records = edit.getRecords();
+        ASSERT_EQ(records.size(), 1);
+        ASSERT_EQ(records[0].entry.file_id, 2);
+        ASSERT_EQ(getTotalStatsNum(blob_store.blob_stats.getStats()), 2);
+
+        // remove one shot blob file
+        blob_store.remove({entry_from_write2});
+        ASSERT_EQ(getTotalStatsNum(blob_store.blob_stats.getStats()), 1);
+    }
+}
+CATCH
+
+TEST_F(BlobStoreTest, TestBigBlobGC)
+try
+{
+    const auto file_provider = DB::tests::TiFlashTestEnv::getContext().getFileProvider();
+    PageId page_id1 = 50;
+    PageId page_id2 = 51;
+    PageId page_id3 = 52;
+
+    BlobStore::Config config_with_small_file_limit_size;
+    config_with_small_file_limit_size.file_limit_size = 800;
+
+    PageEntryV3 entry_from_write1;
+    PageEntryV3 entry_from_write2;
+    PageEntryV3 entry_from_write3;
+    auto blob_store = BlobStore(getCurrentTestName(), file_provider, delegator, config_with_small_file_limit_size);
+    {
+        size_t size_100 = 100;
+        size_t size_500 = 500;
+        size_t size_200 = 200;
+        char c_buff1[size_100];
+        char c_buff2[size_500];
+        char c_buff3[size_200];
+
+        WriteBatch wb;
+        ReadBufferPtr buff1 = std::make_shared<ReadBufferFromMemory>(const_cast<char *>(c_buff1), size_100);
+        ReadBufferPtr buff2 = std::make_shared<ReadBufferFromMemory>(const_cast<char *>(c_buff2), size_500);
+        ReadBufferPtr buff3 = std::make_shared<ReadBufferFromMemory>(const_cast<char *>(c_buff3), size_200);
+        wb.putPage(page_id1, /* tag */ 0, buff1, size_100);
+        wb.putPage(page_id2, /* tag */ 0, buff2, size_500);
+        wb.putPage(page_id3, /* tag */ 0, buff3, size_200);
+        auto edit = blob_store.write(wb, nullptr);
+        const auto & records = edit.getRecords();
+        ASSERT_EQ(records.size(), 3);
+        entry_from_write1 = records[0].entry;
+        entry_from_write2 = records[1].entry;
+        entry_from_write3 = records[2].entry;
+
+        ASSERT_TRUE(blob_store.blob_stats.blobIdToStat(1)->isNormal());
+    }
+
+    config_with_small_file_limit_size.file_limit_size = 400;
+    config_with_small_file_limit_size.heavy_gc_valid_rate = 0.99;
+    {
+        blob_store.reloadConfig(config_with_small_file_limit_size);
+
+        Poco::File file1(blob_store.getBlobFile(1)->getPath());
+        ASSERT_EQ(file1.getSize(), 800);
+        ASSERT_TRUE(blob_store.blob_stats.blobIdToStat(1)->isNormal()); // BlobStat type doesn't change after reload
```

Comment:
So the stat_id=1 could be chosen for later write after `blob_store.reload`? What about keeping the change that do NOT reload the file_size_limit once created?

Reply:
Currently there is no config for `file_limit_size`, so it cannot be reloaded by the customer. Do you mean don't reload it when calling `reloadConfig`?

Reply:
Or we just add some comment about this situation at `BlobStore::reloadConfig`.

Reply:
Add some comment about it.